src/Entities/DreamStuff/DreamDashCollider.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.Xna.Framework;
+using Mono.Cecil.Cil;
+using Monocle;
+using MonoMod.Cil;
+using MonoMod.Utils;
+using System;
+
+namespace Celeste.Mod.CommunalHelper.Entities {
+    [Tracked]
+    class DreamDashCollider : Component {
+        public static readonly Color ActiveColor = Color.Teal;
+        public static readonly Color InactiveColor = Calc.HexToColor("044f63"); // darker teal
+
+        public Collider Collider;
+        public DreamBlockDummy Dummy;
+
+        public Action<Player> OnExit;
+
+        public DreamDashCollider(Collider collider, Action<Player> onExit_player_dreamJumped = null)
+            : base(active: true, visible: false) {
+            Collider = collider;
+            Dummy = new DreamBlockDummy(Entity);
+            OnExit = onExit_player_dreamJumped;
+        }
+
+        public override void Added(Entity entity) {
+            base.Added(entity);
+            Dummy.Entity = entity;
+        }
+
+        /// <summary>
+        /// Checks if the player is colliding with this component.
+        /// </summary>
+        /// <param name="player">The player instance.</param>
+        private bool Check(Player player) {
+            if (Active && Entity != null &&
+                player.GetData().Data.TryGetValue(Player_canEnterDreamDashCollider, out object canEnter) && canEnter.Equals(true)) {
+                
+                Collider collider = Entity.Collider;
+
+                Entity.Collider = Collider;
```

Comment:
this should only be done if `Collider` is not `null`

Reply:
are you sure this is needed? if the entity doesn't have a collider, then there is no collision. i'll just add the check so that it doesn't bother doing it anyway